src/main/java/seedu/address/model/ModelHistoryManager.java

```diff
@@ -230,8 +230,16 @@ public boolean canRedo() {
      * @throws AlfredModelHistoryException
      */
     public ModelHistoryRecord redo() throws AlfredModelHistoryException {
-        //TODO: Update this in v1.3-1.4
-        throw new AlfredModelHistoryException("Not yet implemented");
+        if (this.canRedo()) {
+            int currentIndex = this.history.indexOf(this.current);
+            this.current = this.history.get(currentIndex + 1);
+            ParticipantList.setLastUsedId(this.current.getParticipantListLastUsedId());
+            MentorList.setLastUsedId(this.current.getMentorListLastUsedId());
+            TeamList.setLastUsedId(this.current.getTeamListLastUsedId());
+            return this.current;
```

Comment:
This works right? Mutating a static variable too much is dangerous.

Reply:
Yup you have a good point. Changed this in my latest PR. Now the only point at which the lastUsedId for the EntityLists is set is in ModelManager.